ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/highLevelHumanoidControl/HumanoidWholeBodyControllerCoreManager.java

```diff
@@ -0,0 +1,169 @@
+//      this.controllerCoreCommandDataHolder = controllerCoreCommandDataHolder;
+      this.desiredMPCControllerOutput = desiredMPCControllerOutput;
+
+      controlledJoint = HighLevelHumanoidControllerToolbox.computeJointsToOptimizeFor(fullRobotModel, jointsToIgnore);
+
+      // TODO will be worked later
+//      controllerCoreCommand.setControllerCoreMode(controllerCoreCommandDataHolder.getControllerCoreMode());
+
+      OneDoFJointBasics[] controlledOneDoFJoints = MultiBodySystemTools.filterJoints(controlledJoint, OneDoFJointBasics.class);
+      yoLowLevelOneDoFJointDesiredDataHolder = new YoLowLevelOneDoFJointDesiredDataHolder(controlledOneDoFJoints, registry);
+      yoDesiredMPCControllerOutputDataHolder = new YoLowLevelOneDoFJointDesiredDataHolder("MPCControllerOutput", controlledOneDoFJoints, registry);
+
+   }
+
+   private final YoGraphicListDefinition scs2AdditionalYoGraphics = new YoGraphicListDefinition();
+
+   @Override
+   public YoGraphicDefinition getSCS2YoGraphics()
+   {
+      YoGraphicGroupDefinition group = new YoGraphicGroupDefinition(getClass().getSimpleName());
+      group.addChild(scs2AdditionalYoGraphics);
+      //      WholeBodyControllerCoreFactory wholeBodyControllerCoreFactory = controllerFactoryHelper.getWholeBodyControllerCoreFactory();
+      //      WholeBodyControllerCore wholeBodyControllerCore = wholeBodyControllerCoreFactory.getWholeBodyControllerCore();
+      //      if (wholeBodyControllerCore != null)
+      //         group.addChild(wholeBodyControllerCore.getSCS2YoGraphics());
+      //      LinearMomentumRateControlModule linearMomentumRateControlModule = wholeBodyControllerCoreFactory.getLinearMomentumRateControlModule();
+      //      if (linearMomentumRateControlModule != null)
+      //         group.addChild(linearMomentumRateControlModule.getSCS2YoGraphics());
+      //      group.addChild(controllerFactoryHelper.getHighLevelHumanoidControllerToolbox().getSCS2YoGraphics());
+      //      group.addChild(controllerFactoryHelper.getManagerFactory().getSCS2YoGraphics());
+
+      return group;
+   }
+
+   @Override
+   public void doControl()
+   {
+//      controllerCoreCommand.set(controllerCoreCommandDataHolder);
+
+      if (controllerCoreCommand.getControllerCoreMode() == null)
+         controllerCoreCommand.setControllerCoreMode(WholeBodyControllerCoreMode.INVERSE_DYNAMICS);
+
+      try
+      {
+         // This should be pass by some kind of CASE statements.
+         // Not all the highLevelControllerState use the WholeBodyControllerCore
+         // But WholeBodyControllerCore updates the joint input here.
+         // So, the output of other highLevelStates should be called in here to be delivered to copyJointDesiredsToJoints
+         yoDesiredMPCControllerOutputDataHolder.overwriteWith(desiredMPCControllerOutput);
+         controllerCore.submitControllerCoreCommand(controllerCoreCommand);
+         //         controllerCore.compute();
```

Comment:
how does this work if you're not calling compute?

Reply:
Currently, only the MPCController Output which comes from the controllerThread is used to populate to the actuator. 
`mpcController.getMPCSolution()` in MPCControllerState are saved to desiredMPCCOntrollerOutput in `MPCHumanoidHighLevelControllerManager.copyJointDesiredsToJoints`. And, this is delivered to wholebodycontrollercoreThread by HumanoidRobotMPCContextData. 
Finally, desiredMPCControllerCoreOutput is saved to lowLevelControllerOutput in the copyJointDesiredsToJoints in `HumanoidWholeBodyControllerCoreManager`